### PR TITLE
8332109: Convert remaining tests using com.sun.tools.classfile to ClassFile API

### PR DESCRIPTION
**Pull request by liach**
Some tests are not migrated to the ClassFile API in previous migrations.

 - Some are simple oversights that didn't remove usages of com.sun.tools.classfile;
 - The CallerSensitive ones used an old utility, replaced by CF API-based new code;
 - many in javac are because the files are compiled with older source compatibility. Those patches are converted to have the source code stored in text blocks and compiled within tests using `ToolBox#writeJavaFiles` and `CompilerUtils.compile`;
 - As described in the JBS issue, there are a few other tests not covered; one is in #19193 while the others are blocked by CreateSymbols migration or bugs.

Testing: all modified tests pass.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332109](https://bugs.openjdk.org/browse/JDK-8332109): Convert remaining tests using com.sun.tools.classfile to ClassFile API (**Sub-task** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19206/head:pull/19206` \
`$ git checkout pull/19206`

Update a local copy of the PR: \
`$ git checkout pull/19206` \
`$ git pull https://git.openjdk.org/jdk.git pull/19206/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19206`

View PR using the GUI difftool: \
`$ git pr show -t 19206`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19206.diff">https://git.openjdk.org/jdk/pull/19206.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19206#issuecomment-2106169291)

### CODE REVIEW

**Review by jonathan-gibbons — COMMENTED** [review]
javadoc test change looks OK.

File: test/jdk/jdk/internal/reflect/CallerSensitive/CheckCSMs.java [hunk 1]
```diff
@@ -111,170 +114,162 @@ public static void main(String[] args) throws Exception {
...
-                    return cpref.getClassName().equals(classname) && nat.getName().equals(method);
-                } catch (ConstantPoolException ex) {
-                    throw new RuntimeException(ex);
+        for (var method : clazz.methods()) {
+            var code = method.code().orElse(null);
+            if (code == null)
+                continue;
+
+            boolean needsCsm = false;
+            for (var element : code) {
```
Comment (asotona) [inline test/jdk/jdk/internal/reflect/CallerSensitive/CheckCSMs.java:151]: Scanning the instructions is a bit different approach than in the original test.

File: test/jdk/jdk/internal/reflect/CallerSensitive/CallerSensitiveFinder.java [hunk 2]
```diff
@@ -87,75 +91,83 @@ public static void main(String[] args) throws Exception {
...
+            if (code == null)
+                continue;
+
+            boolean needsCsm = false;
+            for (var element : code) {
+                if (element instanceof InvokeInstruction invoke
+                        && invoke.opcode() == Opcode.INVOKESTATIC
+                        && invoke.method() instanceof MethodRefEntry ref
+                        && ref.owner().name().equalsString(className)
+                        && ref.name().equalsString(methodName)) {
```
Comment (asotona) [inline test/jdk/jdk/internal/reflect/CallerSensitive/CallerSensitiveFinder.java:125]: Is this additional test necessary, I don't see it in the original test.

File: test/langtools/tools/javac/7166455/CheckACC_STRICTFlagOnclinitTest.java [hunk 3]
```diff
@@ -25,98 +25,84 @@
  * @test
  * @bug 7166455
  * @summary javac doesn't set ACC_STRICT bit on <clinit> for strictfp class
- * @modules jdk.jdeps/com.sun.tools.classfile
- * @compile -source 16 -target 16 CheckACC_STRICTFlagOnclinitTest.java
```
Comment (asotona) [inline test/langtools/tools/javac/7166455/CheckACC_STRICTFlagOnclinitTest.java:29]: This compilation cannot be removed and conversion of this test must wait until Class-File API leaves the preview.

File: test/langtools/tools/javac/7166455/CheckACC_STRICTFlagOnclinitTest.java [hunk 4]
```diff
@@ -25,98 +25,84 @@
...
-              "CheckACC_STRICTFlagOnclinitTest$1Foo.class",
-              "CheckACC_STRICTFlagOnclinitTest$1Foo$Bar.class",
-              "CheckACC_STRICTFlagOnclinitTest$1Any.class");
-        if (errors.size() > 0) {
+            throws IOException {
+        Path in = Path.of("in");
+        Path out = Path.of("out");
+        ToolBox toolBox = new ToolBox();
+        toolBox.writeJavaFiles(in, SOURCE);
+        CompilerUtils.compile(in, out, "--release", "16");
```
Comment (asotona) [inline test/langtools/tools/javac/7166455/CheckACC_STRICTFlagOnclinitTest.java:81]: Smart move 👍

File: test/jdk/jdk/internal/reflect/CallerSensitive/CallerSensitiveFinder.java (hunk 2 above)
Comment (liach) [reply test/jdk/jdk/internal/reflect/CallerSensitive/CallerSensitiveFinder.java:125]: This was part of the now-removed ReferenceFinder logic; it scans cp as a fast path and then does instruction scan for all methods to find relevant methods to pass to visitor.

**Review by asotona — APPROVED** [review]
Looks good to me, great job, thanks!

**Comment by liach** [comment]
Thanks for the review!

/integrate

**Pull request closed without merging**